### PR TITLE
backport: merge bitcoin#23280 (coalesce chainstate loading sequence between {,non-}unittest codepaths)

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

* Dependent on https://github.com/dashpay/dash/pull/6296

* Dependent on https://github.com/dashpay/dash/pull/6443

* As one of the backport's intentions were to unify code between `init.cpp` and `setup_common.cpp`, Dash-specific initialization code (to the extent that it can be moved out non-disruptively) has been spun out into `DashChainstateSetup{,Close}()` so it can also be used in `setup_common.cpp` and `validation_chainstatemanager_tests.cpp`.

  This is also why `DashTestSetup{,Close}()` (now `DashPostChainstateSetup{,Close}()`) was introduced in [dash#5531](https://github.com/dashpay/dash/pull/5531).
  * `DashChainstateSetup{,Close}()` (as defined in `node/chainstate.cpp`) cannot take `NodeContext` in because  `node/chainstate.cpp` is used in `bitcoin-chainstate`, which doesn't include `NodeContext` ([source](https://github.com/bitcoin/bitcoin/pull/24304/files#diff-4cb884d03ebb901069e4ee5de5d02538c40dd9b39919c615d8eaa9d364bbbd77R795-R798)), this is reflected by neither `LoadChainstate` nor `VerifyLoadedChainstate` taking in `NodeContext`.

  * To make it less onerous to use in unit tests, `DashChainstateSetup{,Close}()` has been overloaded with a variant that accepts `NodeContext`.

* To remove `LogPrintf` usage in `node/chainstate.cpp`, index enablement reporting has been pulled out of chainstate loading and BLS scheme reporting has been abstracted out using `notify_bls_state`.

## Breaking Changes

None expected.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation **(note: N/A)**
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
LGTM, light ACK 1a0d1256968d79af39b57de98e9a9612c2ee78cb

nit: clang format complains a bit

**Comment by kwvg** [comment]
Current approach of allowing `NodeContext` seems to cause problems down the line, marking as draft until resolved

File: src/node/chainstate.cpp [hunk 1]
```diff
@@ -0,0 +1,329 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <node/chainstate.h>
+
+#include <consensus/params.h>
+#include <deploymentstatus.h>
+#include <node/blockstorage.h>
+#include <node/context.h>
```
Comment (UdjinM6) [inline src/node/chainstate.cpp:10]:
```suggestion
```

**Review by UdjinM6 — APPROVED** [review]
utACK 04523fcd6caf5a3b89c34efb3c15bdbdc9800b89

one tine issue, see below

File: src/node/chainstate.cpp (hunk 1 above)
Comment (kwvg) [reply src/node/chainstate.cpp]: Resolved in latest push

**Review by UdjinM6 — APPROVED** [review]
utACK 19746513b168f210a784dc772d1488f93cb4a2e3

**Pull request merged**